### PR TITLE
Login page, protected routes, log out modals

### PR DESCRIPTION
**Pull request by AlexandruFercal01**

### CODE REVIEW

File: src/components/LoginPage/LoginForm.tsx [hunk 1]
```diff
@@ -18,51 +20,65 @@ interface Login {
...
     password,
     handleUsername,
     handlePassword,
 }) => {
     const navigate = useNavigate();
-    const [authenticated, setAuthenticated] = useState(
-        localStorage.getItem("authenticated")
-    );
+    const [users, setUsers] = useState<userInfo[]>();
+    const getUserss = async () => {
```
Comment (CatalinTiu) [inline src/components/LoginPage/LoginForm.tsx:31]: Maybe rename getUserss to something less vague

File: src/components/Admin/LogOutModal/LogOutModal.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Box, IconButton, Modal } from "@mui/material";
+import React from "react";
+
+import "../DeleteConfirmationModal/DeleteConfirmationModal.css";
+import HighlightOffIcon from "@mui/icons-material/HighlightOff";
+
+type ConfirmationModalProps = {
```
Comment (CatalinTiu) [inline src/components/Admin/LogOutModal/LogOutModal.tsx:7]: Change Type to Interface

**Review by CatalinTiu — CHANGES_REQUESTED** [review]

**Review by CatalinTiu — APPROVED** [review]

File: src/api/users.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+import { userInfo } from "../interfaces/interfaces";
+
+const getUsers = async (): Promise<userInfo[]> => {
+    const result = await axios.get("http://localhost:3001/users");
+    return result.data;
+};
+
+const getUser = async (
```
Comment (daniel-mita) [inline src/api/users.ts:9]: This can be easily mistaken with getUsers, rename this to getSingleUser

File: src/components/Admin/Admin.tsx [hunk 4]
```diff
@@ -169,6 +174,13 @@ const Admin = () => {
                 }
                 roomTitle={deleteRoomTitle}
             />
+            <LogoutConfirmationModal
+                open={showLogoutModal}
+                onClose={() => {
+                    handleClose(setShowLogoutModal);
```
Comment (daniel-mita) [inline src/components/Admin/Admin.tsx:180]: Why pass the setter function here as an argument when you already have access to it in the function scope?

File: src/components/Admin/LogOutModal/LogOutModal.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    onSubmit,
+}: ConfirmationModalProps) => {
+    return (
+        <Modal open={open} onClose={onClose}>
+            <Box className="confirmation-modal">
+                <Box
+                    sx={{
+                        display: "flex",
+                        justifyContent: "space-between",
+                        //alignItems: "left",
```
Comment (daniel-mita) [inline src/components/Admin/LogOutModal/LogOutModal.tsx:20]: Delete this comment if it's not relevant

File: src/components/LoginPage/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    }, []);
+
+    useEffect(() => {
+        localStorage.setItem("authenticated", "false");
+        localStorage.setItem("user_type", "none");
+    }, []);
+
+    const handleSubmit = (e: any) => {
+        e.preventDefault();
+        console.log(users);
```
Comment (daniel-mita) [inline src/components/LoginPage/LoginForm.tsx:45]: Remove this console.log

File: src/components/LoginPage/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    };
+
+    useEffect(() => {
+        getUsersAccounts();
+    }, []);
+
+    useEffect(() => {
+        localStorage.setItem("authenticated", "false");
+        localStorage.setItem("user_type", "none");
+    }, []);
```
Comment (daniel-mita) [inline src/components/LoginPage/LoginForm.tsx:41]: Every time the component rerenders this block of code will run, are you sure you want this behaviour?

File: src/interfaces/interfaces.ts [hunk 8]
```diff
@@ -184,3 +185,27 @@ export interface dateSelectorValid {
 export interface iQuickBookGlobal {
     onClickQuickBookGlobal: () => void;
 }
+
+export interface userInfo{
```
Comment (daniel-mita) [inline src/interfaces/interfaces.ts:189]: interfaces start with capital letter

File: src/components/TabletApp/LogOutModal.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+                    <Box
+                        sx={{
+                            display: "flex",
+                            justifyContent: "flex-end",
+                            paddingRight: 5,
+                        }}
+                    >
+                        <Button
+                            onClick={() => {
+                                navigate("/login");
```
Comment (daniel-mita) [inline src/components/TabletApp/LogOutModal.tsx:43]: If you only navigate to the login screen, the user will still be logged in, extract this in a sepparate function and handle all the effects of the logout like deleting tokens and whatever from the local storage.

**Review by daniel-mita — CHANGES_REQUESTED** [review]

**Review by robert-szalkai — APPROVED** [review]

**Pull request merged**